test/search.cc

```diff
@@ -162,7 +173,9 @@ void search(valhalla::baldr::Location location,
   PathLocation::toPBF(location, &pbf, reader);
   location = PathLocation::fromPBF(pbf);
 
-  const auto results = Search({location}, reader);
+  const auto costing = create_costing();
+
+  const auto results = Search({location}, reader, costing);
```

Comment:
Second segfault was due to `costing` being initialized as default value to `{}`. I removed the default value and supplied a costing object to `Search`.

Reply:
that will drastically change the tests though. costing is not required for edge search. if we are segfaulting because of it we need to fix it down stream not here.